custom_components/fellow/fellow_aiden/__init__.py

```
@@ -194,287 +255,315 @@ def __device(self):
         self._log.debug("Brew profile passed checks")
         profile_url = self.BASE_URL + self.API_PROFILES.format(id=self._brewer_id)
-        response = self._request_with_reauth("post", profile_url, json=data)
-        self._ensure_success(response, "Profile creation")
+        response = await self._request_with_reauth("post", profile_url, json=data)
+        await self._ensure_success(response, "Profile creation")
 
-        parsed = self._parse_response(response)
+        parsed = await self._parse_response(response)
         if "id" not in parsed:
-            raise Exception("Error in processing: %s" % parsed)
+            raise Exception(f"Error in processing: {parsed}")
```

Ruff (0.15.2)

[warning] Create your own exception

(TRY002)